$NetBSD: patch-configure_in,v 1.4 2021/09/13 11:37:34 dsainty Exp $

AM_CONFIG_HEADER is obsolete.

Mac OS X has a normal include directory if the JPATH is set appropriately.

UTS_RELEASE isn't even reliably available on Linux, let alone portable.

Recognise Java 1.6, and openjdk7 as 1.7.

The printer port support is experimental, and only supported on some
platforms.  Disable it for consistency, using a patch since
--disable-PRINTER only works on Linux.

Force use of Pkgsrc libtool, even on Mac OS X.
 
--- configure.in.orig	2006-01-30 15:37:47.000000000 +1100
+++ configure.in	2017-04-24 19:53:54.000000000 +1000
@@ -17,7 +17,7 @@
 #   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #-------------------------------------------------------------------------
 AC_INIT(src/SerialImp.c)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CANONICAL_SYSTEM
 AC_MSG_WARN(Trying libtool.  If the following fails install libtool)
 AC_PROG_CC
@@ -409,15 +409,7 @@
 		echo "------------------------------------------------------" 1>&2
 	fi;
 }
-[ case $OS_NAME in
-	Mac\ OS\ X)
-        JAVAINCLUDEDIR=$JPATH/../../../Headers
-        ;;
-	*)
-        JAVAINCLUDEDIR=$JPATH/include
-        ;;
-esac ]
-
+JAVAINCLUDEDIR=$JPATH/include
 JAVAINCLUDE="-I\$(JAVAINCLUDEDIR)"
 if test "$CC" = "gcc" || test "$CC"="egcs"; then
 	case $JAVA_VENDOR in
@@ -458,12 +450,12 @@
 [ case $OS_NAME in 
 Linux)
 	LDFLAGS=$LDFLAGS" -lpthread"
-	check_kernel_headers
+	#check_kernel_headers
 	check_java_headers
 	CFLAGS=$CFLAGS" -D__need_timespec"
 	case $JAVA_VERSION in
 
-	1.2*|1.3*|1.4*|1.5*) 
+	1.2*|1.3*|1.4*|1.5*|1.6*|1.7*) 
 		#fix_parameters $JPATH/jre/lib/javax.comm.properties
 		CLASSPATH=".:\$(TOP):\$(TOP)/src:"`find $JPATH/ -name RXTXcomm.jar |head -n1`
 		JHOME=$JPATH"/jre/lib/ext"
@@ -527,13 +519,13 @@
 ;;
 
 *BSD)
-	LDFLAGS=$LDFLAGS" -lc_r"
+	LDFLAGS=$LDFLAGS" -lpthread"
 	JHOME=$JPATH/"lib"
 	CFLAGS=$CFLAGS" -D_NO_POSIX=1 -D_NO_XOPEN4=1"
 	TARGETLIB="\$(target_triplet)/librxtxSerial.la \
 			\$(target_triplet)/librxtxParallel.la"
 	case $JAVA_VERSION in
-	1.2*|1.3*|1.4*|1.5*) 
+	1.2*|1.3*|1.4*|1.5*|1.6*|1.7*) 
 		#fix_parameters $JPATH/jre/lib/javax.comm.properties
 		CLASSPATH=".:\$(TOP):\$(TOP)/src:"`find $JPATH/ -name RXTXcomm.jar |head -n1`
 		RXTX_PATH="\$(JPATH)/jre/lib/\$(OS_ARCH)"
@@ -721,7 +713,7 @@
 	CLASSPATH=".:\$(TOP):\$(TOP)/src:\$(JPATH)/lib/classes.zip:\$(JPATH)/lib/RXTXcomm.jar:$CLASSPATH"
 
 	case $JAVA_VERSION in
-	1.2*|1.3*|1.4*|1.5*)
+	1.2*|1.3*|1.4*|1.5*|1.6*|1.7*)
 		RXTX_PATH="\$(JPATH)/jre/lib/\$(OS_ARCH)"
 	;;
 	1.1*)
@@ -816,6 +808,9 @@
 ;;
 esac ]
 
+TARGETLIB="\$(target_triplet)/librxtxSerial.\$(LIB_EXTENSION)"
+GLIBTOOL="\$(SHELL) \$(TOP)/libtool"
+
 AC_OUTPUT(Makefile)
 
 
